torchvision/models/_api.py

```diff
@@ -203,19 +204,45 @@ def wrapper(fn: Callable[..., M]) -> Callable[..., M]:
     return wrapper
 
 
-def list_models(module: Optional[ModuleType] = None) -> List[str]:
+def list_models(
+    module: Optional[ModuleType] = None,
+    include_filters: Union[Iterable[str], str, None] = None,
+    exclude_filters: Union[Iterable[str], str, None] = None,
+) -> List[str]:
     """
     Returns a list with the names of registered models.
 
     Args:
         module (ModuleType, optional): The module from which we want to extract the available models.
+        include_filters (str or Iterable[str], optional): Filter(s) for including the models from the set of all models.
+            Filters are passed to fnmatch to match Unix shell-style wildcards. In case of many filters, the results is
+            the union of individual filters.
+        exclude_filters (str or Iterable[str], optional): Filter(s) applied after include_filters to remove models.
+            Filter are passed to fnmatch to match Unix shell-style wildcards. In case of many filters, the results is
+            removal of all the models that match any individual filter.
 
     Returns:
         models (list): A list with the names of available models.
     """
-    models = [
+    all_models = [
         k for k, v in BUILTIN_MODELS.items() if module is None or v.__module__.rsplit(".", 1)[0] == module.__name__
     ]
+    if include_filters is not None:
+        models = set()
+        if isinstance(include_filters, str):
+            include_filters = [include_filters]
+        for include_filter in include_filters:
+            models = models | set(fnmatch.filter(all_models, include_filter))
+    else:
+        models = all_models
+
+    if exclude_filters is not None:
+        if isinstance(models, list):
+            models = set(models)
```

Comment:
Might as well do it regardless and save some brain-overhead. It's also a tiiiiny bit safer, as it makes this block independent of the logic of the previous block.

```suggestion
        models = set(models)
```